### PR TITLE
Prepare the project structure for Scorm 2004 API addition

### PR DESCRIPTION
**Pull request by xob**
While no functionality has been changed, this pull request is a major overhaul of the code structure.

I did previously warn you that I would come in with a pretty massive PR. While this one does not add the Scorm 2004 API yet, it does lay the groundwork that will allow me to easily add that API to the project.

I am well aware that this is a lot to take in, especially on a monday morning. I am open to discussion, suggestions, etc.

Major changes:
- Split functionality in different files
- Added Grunt to build the single output file, as well as a minified output file
- Changed code styling standards to fit my own personal style, and added ESLint to ensure these standards are maintained in the future
  - Changed indentation from 4 spaces to 2 spaces
  - Collapsed if-else blocks a little (no newline between `}` and `else`. etc.)
  - Grouped object properties with their getter and setter functions
  - Collapsed getter and setter functions to single lines where possible, using ternary operators when relevant

To build the project, a first `npm install` needs to be done, then:
- (optional) Increment project version in the `package.json` file
- run `grunt`

### CODE REVIEW

File: build/scormAPI.js [hunk 1]
```diff
@@ -0,0 +1,935 @@
+/*! simplify-scorm v1.0.1 - 2018-04-09 */
```
Comment (gabrieldoty) [inline build/scormAPI.js]: Thinking the build folder should not be included

Comment (xob) [reply build/scormAPI.js]: I could exclude it. I'm not too familiar with github's release feature, but I assume then that it allows you to supply the built files for people to download later?

I wouldn't want people who just want the built files to have to download the source code and build it themselves.

Comment (gabrieldoty) [reply build/scormAPI.js]: GitHub has an option to attach a binary as part of the official release, I am pretty sure we can use this to attach the minified final output to a release, but I will also verify this

Comment (xob) [reply build/scormAPI.js]: Ah, yes, I see that now. Removing the build folder.

**Pull request merged**